scripts/quests/outlands/Black_Market.lua

```diff
@@ -0,0 +1,139 @@
+                    return quest:event(16)
+                end,
+
+                onTrade = function(player, npc, trade)
+                    if npcUtil.tradeHasExactly(trade, { { xi.item.NORTHERN_FUR, 4 } }) then
+                        return quest:progressEvent(17, xi.item.NORTHERN_FUR, xi.item.NORTHERN_FUR)
+                    elseif npcUtil.tradeHasExactly(trade, { { xi.item.PIECE_OF_EASTERN_POTTERY, 4 } }) then
+                        return quest:progressEvent(18, xi.item.PIECE_OF_EASTERN_POTTERY, xi.item.PIECE_OF_EASTERN_POTTERY)
+                    elseif npcUtil.tradeHasExactly(trade, { { xi.item.SOUTHERN_MUMMY, 4 } }) then
+                        return quest:progressEvent(19, xi.item.SOUTHERN_MUMMY, xi.item.SOUTHERN_MUMMY)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [17] = function(player, csid, option, npc)
+                    if quest:complete(player) then
+                        player:confirmTrade()
+                        player:addFame(xi.fameArea.NORG, 40)
+                        npcUtil.giveCurrency(player, 'gil', 1500)
+                    end
+                end,
+
+                [18] = function(player, csid, option, npc)
+                    if quest:complete(player) then
+                        player:confirmTrade()
+                        player:addFame(xi.fameArea.NORG, 50)
+                        npcUtil.giveCurrency(player, 'gil', 2000)
+                    end
+                end,
+
+                [19] = function(player, csid, option, npc)
+                    if quest:complete(player) then
+                        player:confirmTrade()
+                        player:addFame(xi.fameArea.NORG, 80)
+                        npcUtil.giveCurrency(player, 'gil', 3000)
+                    end
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_COMPLETED
+        end,
+
+        [xi.zone.NORG] =
+        {
+            ['Muzaffar'] =
+            {
+                onTrigger = function(player, npc)
+                    return quest:event(20)
+                end,
+
+                onTrade = function(player, npc, trade)
+                    if npcUtil.tradeHasExactly(trade, { { xi.item.NORTHERN_FUR, 4 } }) then
+                        return quest:progressEvent(17, 1199, 1199)
+                    elseif npcUtil.tradeHasExactly(trade, { { xi.item.PIECE_OF_EASTERN_POTTERY, 4 } }) then
+                        return quest:progressEvent(18, 1200, 1200)
+                    elseif npcUtil.tradeHasExactly(trade, { { xi.item.SOUTHERN_MUMMY, 4 } }) then
+                        return quest:progressEvent(19, 1201, 1201)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [17] = function(player, csid, option, npc)
```

Comment:
While not overly necessary, there's a lot of repeated code here, which could probably be tabled

Reply:
Can do! - let me know if I should nix the quest reward title and just throw it into the new handleTurnIn method.